build.gradle.kts

```diff
@@ -200,6 +197,16 @@ kotlin.target.compilations {
   }
 }
 
+// Just override flags for plugin sources, we can still use new features in tests.
+tasks.compileKotlin {
+  compilerOptions {
+    // https://docs.gradle.org/current/userguide/compatibility.html#kotlin
+    @Suppress("DEPRECATION") // TODO: bump apiVersion to 2.0 to match Gradle 9.0
+    apiVersion = KotlinVersion.KOTLIN_1_8
+    languageVersion = apiVersion
```

Comment:
When we go back to 1.8, something is wrong in the tests

```
* What went wrong:	
A problem occurred configuring project ':main'.	
> class com/github/jengelman/gradle/plugins/shadow/tasks/ShadowJar: inline fun relocate(): compile avoidance is not supported with public inline functions	
```

https://scans.gradle.com/s/jisjgaaam5e22/tests/overview?outcome=FAILED

Reply:
CC @wolfs

If we can bump the lang level to `KOTLIN_2_0`? Everything works well after that.